pkg/controller/qdrouterd/qdrouterd_controller.go

```diff
@@ -321,6 +328,34 @@ func (r *ReconcileQdrouterd) Reconcile(request reconcile.Request) (reconcile.Res
 		return reconcile.Result{}, err
 	}
 
+	// create route for exposed listeners
+	for _, listener := range instance.Spec.Listeners {
+		if listener.Expose {
```

Comment:
Could perhaps check that listener is either http or ssl enabled and warn if not (since route will not work apart from those cases)?